### ISSUE TITLE
embedding层开启sparse=True后，无法查看该层权重的梯度。

### ISSUE THREAD

**Issue by ZACP-DERMATOLOGY** [PaddlePaddle/Paddle#69992]
### bug描述 Describe the Bug

description：paddle.nn.embedding层开启sparse=True后无法查看该层权重的梯度了，显示张量未初始化。但是我使用embedding.weight.gradient()可以勉强显示，但是paddle框架的输出结果提示了“该.gradient()马上就要在未来被移除了，尽快改为使用.grad用于查看参数”。而我使用.grad查看该层权重梯度的话就会提示我下面输出结果所示的内容，也就是张量未初始化。
我该如何查看开启sparse=True时的embedding层权重梯度呢？

Running code：
```
import paddle
import paddle.nn as nn

embedding=nn.Embedding(3,2,sparse=True)
embedding.weight.set_value(paddle.arange(6).reshape((3,2)).astype('float32'))
data=paddle.to_tensor((0,1))
out=embedding(data)
y=out.sum()*2
y.backward()
print(embedding.weight.grad) #这行还可以改为print(embedding.weight.gradient())
```
output result：
```
W1201 12:24:43.693264  4996 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 7.5, Driver API Version: 12.6, Runtime API Version: 12.3
W1201 12:24:43.694262  4996 gpu_resources.cc:164] device: 0, cuDNN Version: 9.0.
Tensor(Not initialized)
(.venv)
```

### 其他补充信息 Additional Supplementary Information

感谢技术大佬花费解决问题我的issue，不胜感激。

**Comment by chen2016013** [comment]
使用embedding.weight.gradient()的打印结果是什么样的？

**Comment by chen2016013** [comment]
![image](https://github.com/user-attachments/assets/114ead17-5d89-476b-a3a6-24cd4e9f1620)
应该是我们的打印相关代码里存在bug，可以先使用np.array()转化后打印，或者打印.rows()查看。我们后续会尽快修复打印问题。

**Comment by chen2016013** [comment]
相关代码：https://github.com/PaddlePaddle/Paddle/blob/973924ca3a2b230a1cc0393a22731023878f3b51/python/paddle/base/dygraph/tensor_patch_methods.py#L373

**Comment by ZACP-DERMATOLOGY** [comment]
多谢多谢，我还担心那个api以后看不了梯度了
感谢大佬😘

**Comment by chen2016013** [comment]
https://github.com/PaddlePaddle/Paddle/pull/70017
修复代码已提交，欢迎关注。感谢您对开源社区的支持！

**Comment by ZACP-DERMATOLOGY** [comment]
十分感谢技术大牛，祝paddle越来越好！








------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月6日(星期五) 下午5:28
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PaddlePaddle/Paddle] embedding层开启sparse=True后，无法查看该层权重的梯度。 (Issue #69992)














 
#70017
 修复代码已提交，欢迎关注。感谢您对开源社区的支持！
